src/java.base/share/classes/java/io/ObjectInputFilter.java

```diff
@@ -214,92 +397,175 @@
+                try {
+                    // Load using the system class loader, the named class may be an application class.
+                    // The static initialization of the class or constructor may create a race
+                    // if either calls Config.setSerialFilterFactory; the command line configured
+                    // Class should not be overridden.
+                    Class<?> factoryClass= Class.forName(factoryClassName, true,
+                            ClassLoader.getSystemClassLoader());
+                    @SuppressWarnings("unchecked")
+                    BinaryOperator<ObjectInputFilter> f =
+                            (BinaryOperator<ObjectInputFilter>)
+                            factoryClass.getConstructor().newInstance(new Object[0]);
+                    if (serialFilterFactory != null) {
+                        configLog.log(System.Logger.Level.ERROR,
+                                "FilterFactory provided on the command line can not be overridden");
+                        // Do not continue if configuration not initialized
+                        throw new ExceptionInInitializerError("FilterFactory provided on the command line can not be overridden");
+                    }
+                    factory = f;
+                } catch (RuntimeException | ClassNotFoundException | NoSuchMethodException |
+                        IllegalAccessException | InstantiationException | InvocationTargetException ex) {
+                    configLog.log(System.Logger.Level.ERROR,
+                            "Error configuring filter factory", ex);
+                    // Do not continue if configuration not initialized
+                    throw new ExceptionInInitializerError(ex);
+                }
+            } else {
+                factory = new BuiltinFilterFactory();
+            }
+            serialFilterFactory = factory;
 
             // Setup shared secrets for RegistryImpl to use.
             SharedSecrets.setJavaObjectInputFilterAccess(Config::createFilter2);
         }
 
         /**
-         * Current configured filter.
+         * Config has no instances.
+         */
+        private Config() {
+        }
+
+        /**
+         * Logger for debugging.
          */
-        private static volatile ObjectInputFilter serialFilter = configuredFilter;
+        private static void filterLog(System.Logger.Level level, String msg, Object... args) {
+            if (configLog != null) {
+                configLog.log(level, msg, args);
+            }
+        }
 
         /**
-         * Returns the system-wide serialization filter or {@code null} if not configured.
+         * Returns the static JVM-wide deserialization filter or {@code null} if not configured.
          *
-         * @return the system-wide serialization filter or {@code null} if not configured
+         * @return the static JVM-wide deserialization filter or {@code null} if not configured
```

Comment:
Is "static" significant here? Can it be dropped?   I fine myself questioning if the "static JVM-wide" and "JVM-wide" are two different filters. If we do this then we have just two terms: 1) the "JVM-wide deserialization filter" and 2) the "JVM-wide deserialization filter factory".

Additionally, can you please check all occurrence of these, to ensure that they are used consistently in all parts of the spec. I think I see serial/serialization (without the "de" ) used in a few places.

Reply:
The static is intended to distinguish that single filter from the others.  The static vs current distinction is part of JEP 290 from which this evolved.  The migration to "de-serialization" from the previous "serialization" is as yet incomplete.